### DISCUSSION TITLE
Mount and manage images from within the host?

### DISCUSSION THREAD

**Discussion by nielsmh** [ZuluSCSI/ZuluSCSI-firmware#370] [Ideas]
I don't know a whole lot about the SCSI protocol and what is possible and not, so this may be impossible.

But it would be really neat to have a way to manage disk images from within the system ZuluSCSI is connected to. If it could be possible to implement custom SCSI commands to list, mount and eject image files, and then have utilities for various host systems (DOS, Amiga, Mac, etc.) to invoke those commands.

For writable removable media types (floppy, tape, others?), having commands to create new, blank images would also be handy.

**Comment by MeTalOneOEight** [comment]
Have you tried the CD changer eject feature for that purpose?
It's in the config examples.

It should be technically possible to access the SD card itself over SCSI.
MacSD (a more classic Mac aimed SCSI SD) has a Commander app to copy files from/to the SD.

**Reply by nielsmh** [reply]
Yes, I do use the "eject" feature already, and it's very useful when you have just a few images to cycle between. But if you have more images in the cycle, perhaps even 10 or more, it gets rather inconvenient, so some mechanism to select one directly would be nice to have.

**Reply by aperezbios** [reply]
I agree. Another way you can potentially work around this current limitation is to have multiple virtual SCSI CD-ROM drives configured, each with their own directory of image files, but that's still no great solution, as you have to keep track of/know which images are associated with which SCSI ID.

**Comment by nielsmh** [comment]
I see #406 implemented a mechanism to interact with the ZuluSCSI from the host system. The protocol seems to be implemented via a vendor command extension, but not thoroughly documented.

Do anyone know if there are existing implementations (other than the original Toolbox for Mac) to interact with it, or any work is ongoing to make any?

I might look into developing a tool for DOS, but can't promise anything since I'd need to learn a bunch of new tools and APIs first.
